FactoryMethod/src/main/java/com/ben/abstractnotification/NotificationFactory.java

```diff
@@ -0,0 +1,23 @@
+package com.ben.abstractnotification;
+
+public class NotificationFactory extends AbstractFactory {
+    public Notification createNotification(String typeOfNotification)
+    {
+        //This is our factory class/method to instantiate the concrete class
+        if(typeOfNotification == null || typeOfNotification.isEmpty())
+            return null;
```

Comment:
On line 15 you are throwing an exception if the input doesn't match - so I think you can remove this `if` statement. I think it's a lot nicer to throw an exception than return null

Reply:
Makes the code a lot cleaner and more readable so thanks ;)